asm-utils/src/data_storage.rs

```diff
@@ -2,76 +2,157 @@
 
 use std::collections::BTreeMap;
 
-use crate::{data_parser::DataValue, utils::next_multiple_of_four};
+use crate::{
+    data_parser::DataValue,
+    utils::{alignment_size, next_aligned},
+};
 
 pub enum SingleDataValue<'a> {
     Value(u32),
-    LabelReference(&'a String),
-    Offset(&'a String, &'a String),
+    LabelReference(&'a str),
+    Offset(&'a str, &'a str),
```

Comment:
Why str, just to complicate lifetimes? :D

Reply:
I didn't touch the lifetimes. It was already a lifetime annotated reference. I just changed to `str` because it is more general.